src/system/term/mod.rs

```diff
@@ -164,11 +168,22 @@ pub fn current_tty_name() -> io::Result<OsString> {
     std::io::stdin().ttyname()
 }
 
+/// Rust standard library "IsTerminal" is not secure for setuid programs (CVE-2023-2002)
+pub trait IsTerminalForSudo {
+    fn is_terminal(&self) -> bool;
+}
+
+impl<F: AsRawFd> IsTerminalForSudo for F {
+    fn is_terminal(&self) -> bool {
```

Comment:
I think I'd put this under the already existing `Terminal` trait as it is sort of related.

Reply:
Sure thing.